iroh-net/src/hp/derp/http/mesh_clients.rs

```diff
@@ -129,67 +144,82 @@ mod tests {
+            futures::future::try_join_all(server_a_meshed),
+        )
+        .await??;
+        tokio::time::timeout(
+            std::time::Duration::from_secs(5),
+            futures::future::try_join_all(server_b_meshed),
+        )
+        .await??;
+
         let alice_key = SecretKey::generate();
-        tracing::info!("client alice: {:?}", alice_key.public_key());
+        println!("client alice: {:?}", alice_key.public_key());
         let alice = ClientBuilder::new()
             .server_url(a_url)
             .build(alice_key.clone())?;
         let _ = alice.connect().await?;
 
         let bob_key = SecretKey::generate();
-        tracing::info!("client bob: {:?}", bob_key.public_key());
+        println!("client bob: {:?}", bob_key.public_key());
         let bob = ClientBuilder::new()
             .server_url(b_url)
             .build(bob_key.clone())?;
         let _ = bob.connect().await?;
 
-        // needs time for the mesh network to fully mesh
-        // packets may get dropped the first go-around
-        // this will loop until we get the first keepalive, message, which means
-        // there is really something wrong if we can't gt
-        // send bob a message from alice
         let msg = "howdy, bob!";
-        tracing::info!("send message from alice to bob");
+        println!("send message from alice to bob");
         alice.send(bob_key.public_key(), msg.into()).await?;
 
-        loop {
-            tokio::select! {
-                recv = bob.recv_detail() => {
-                    let (recv, _) = recv?;
-                    if let ReceivedMessage::ReceivedPacket { source, data } = recv {
-                        assert_eq!(alice_key.public_key(), source);
-                        assert_eq!(msg, data);
-                        break;
-                    } else {
-                        bail!("unexpected ReceivedMessage {recv:?}");
-                    }
-                }
-                _ = tokio::time::sleep(std::time::Duration::from_millis(100)) => {
-                    tracing::info!("attempting to send another message from alice to bob");
-                    alice.send(bob_key.public_key(), msg.into()).await?;
+        // ensure we get the message, but allow other chatter between the
+        // client and the server
+        let b = bob.clone();
+        let alice_pub_key = alice_key.public_key();
+        tokio::time::timeout(std::time::Duration::from_secs(5), async move {
+            loop {
+                let (recv, _) = b.recv_detail().await?;
+                if let ReceivedMessage::ReceivedPacket { source, data } = recv {
+                    assert_eq!(alice_pub_key, source);
+                    assert_eq!(msg, data);
+                    println!("bob received packet from alice");
+                    return Ok::<(), anyhow::Error>(());
+                } else {
+                    eprintln!("bob received unexpected message {recv:?}");
```

Comment:
did you figure out why there are other messages?

Reply:
yea, every few runs of `test_mesh_network` on the cross-compile binaries, the setup of the test would be slow enough that because we connected to `alice` first but then sent and received messages to it last, we might get a `KeepAlive` that ticked before we sent it a message from `bob`. So instead of failing on a `KeepAlive`, I just time-boxed the `recv_detail`.